In [1]:
import pymongo

In [2]:
# Initialize PyMongo to work with MongoDBs
conn = 'mongodb://localhost:27017'
client = pymongo.MongoClient(conn)

In [3]:
# Define database and collection
db = client.etl_project_db
collection_oc = db.oc_data
collection_la = db.la_data
collection_city = db.city_zipcode
collection_income = db.income_data

In [4]:
# Script to join either of the county's data with city data and income data
join = [{'$lookup':{
            'from': "city_zipcode",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "city_data"}
        },
        {'$unwind':"$city_data" },
        {'$lookup':{
            'from': "income_data",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "income_data"}
        },
        {'$unwind':"$income_data" },
        {   
        '$project':{
            'zipcode' : 1,
            'city_name' : "$city_data.city",
            'total_population' : 1,
            'male_population' : 1,
            'female_population' : 1,
            'total_household' : 1,
            'adjusted_gross_income':"$income_data.adjusted_gross_income"
        } 
    }
]

In [5]:
# Use join script for OC
for data in (db.oc_data.aggregate(join)):
    print(data)

{'_id': ObjectId('5dd0a8a78dd50569ef1ac076'), 'zipcode': '90620', 'total_population': 45113, 'male_population': 22241, 'female_population': 22872, 'total_household': 13268, 'city_name': 'Buena Park', 'adjusted_gross_income': 1019363}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac077'), 'zipcode': '90621', 'total_population': 35153, 'male_population': 17377, 'female_population': 17776, 'total_household': 10304, 'city_name': 'Buena Park', 'adjusted_gross_income': 619992}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac078'), 'zipcode': '90623', 'total_population': 15554, 'male_population': 7516, 'female_population': 8038, 'total_household': 5072, 'city_name': 'La Palma', 'adjusted_gross_income': 472545}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac079'), 'zipcode': '90630', 'total_population': 47993, 'male_population': 23204, 'female_population': 24789, 'total_household': 15785, 'city_name': 'Cypress', 'adjusted_gross_income': 1485267}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac07a'), 'zipcode': '90631', 'total_p

In [6]:
# Use join script for LA
for data in (db.la_data.aggregate(join)):
    print(data)

{'_id': ObjectId('5dd0a89b8dd50569ef1ab96a'), 'zipcode': '90001', 'total_population': 57110, 'male_population': 28468, 'female_population': 28642, 'total_household': 12971, 'city_name': 'Los Angeles', 'adjusted_gross_income': 492447}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab96b'), 'zipcode': '90002', 'total_population': 51223, 'male_population': 24876, 'female_population': 26347, 'total_household': 11731, 'city_name': 'Los Angeles', 'adjusted_gross_income': 438935}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab96c'), 'zipcode': '90003', 'total_population': 66266, 'male_population': 32631, 'female_population': 33635, 'total_household': 15642, 'city_name': 'Los Angeles', 'adjusted_gross_income': 581608}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab96d'), 'zipcode': '90004', 'total_population': 62180, 'male_population': 31302, 'female_population': 30878, 'total_household': 22547, 'city_name': 'Los Angeles', 'adjusted_gross_income': 1451721}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab96e'), 'zipcode': '90005

{'_id': ObjectId('5dd0a89b8dd50569ef1ab9d6'), 'zipcode': '90401', 'total_population': 6722, 'male_population': 3524, 'female_population': 3198, 'total_household': 4188, 'city_name': 'Santa Monica', 'adjusted_gross_income': 735292}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9d7'), 'zipcode': '90402', 'total_population': 12250, 'male_population': 5809, 'female_population': 6441, 'total_household': 5301, 'city_name': 'Santa Monica', 'adjusted_gross_income': 1869235}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9d8'), 'zipcode': '90403', 'total_population': 24525, 'male_population': 11426, 'female_population': 13099, 'total_household': 13970, 'city_name': 'Santa Monica', 'adjusted_gross_income': 1503300}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9d9'), 'zipcode': '90404', 'total_population': 21360, 'male_population': 10292, 'female_population': 11068, 'total_household': 10089, 'city_name': 'Santa Monica', 'adjusted_gross_income': 944501}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9da'), 'zipcode': '90405',

In [7]:
# Script to get top 10 zip with highest population city data and income data
top10 = [
        { '$sort': { "total_population": -1 } },
        { '$limit': 10 },
        {'$lookup':{
            'from': "city_zipcode",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "city_data"}
        },
        {'$unwind':"$city_data" },
        {'$lookup':{
            'from': "income_data",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "income_data"}
        },
        {'$unwind':"$income_data" },
        {   
        '$project':{
            'zipcode' : 1,
            'city_name' : "$city_data.city",
            'total_population' : 1,
            'male_population' : 1,
            'female_population' : 1,
            'total_household' : 1,
            'adjusted_gross_income':"$income_data.adjusted_gross_income"
        } 
    }
]

In [8]:
# Use top 10 population script for OC
for data in (db.oc_data.aggregate(top10)):
    print(data)

{'_id': ObjectId('5dd0a8a78dd50569ef1ac0a6'), 'zipcode': '92683', 'total_population': 89747, 'male_population': 44391, 'female_population': 45356, 'total_household': 26223, 'city_name': 'Westminster', 'adjusted_gross_income': 1715190}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac0ad'), 'zipcode': '92704', 'total_population': 88123, 'male_population': 44951, 'female_population': 43172, 'total_household': 19569, 'city_name': 'Santa Ana', 'adjusted_gross_income': 1251203}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac0b6'), 'zipcode': '92804', 'total_population': 85914, 'male_population': 42316, 'female_population': 43598, 'total_household': 24363, 'city_name': 'Anaheim', 'adjusted_gross_income': 1306972}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac0b7'), 'zipcode': '92805', 'total_population': 70401, 'male_population': 35868, 'female_population': 34533, 'total_household': 17599, 'city_name': 'Anaheim', 'adjusted_gross_income': 1044765}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac07a'), 'zipcode': '90631', 'tot

In [9]:
# Use top 10 population script for LA
for data in (db.la_data.aggregate(top10)):
    print(data)

{'_id': ObjectId('5dd0a89b8dd50569ef1ab9ed'), 'zipcode': '90650', 'total_population': 105549, 'male_population': 52364, 'female_population': 53185, 'total_household': 27130, 'city_name': 'Norwalk', 'adjusted_gross_income': 1673809}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab973'), 'zipcode': '90011', 'total_population': 103892, 'male_population': 52794, 'female_population': 51098, 'total_household': 22168, 'city_name': 'Los Angeles', 'adjusted_gross_income': 808914}
{'_id': ObjectId('5dd0a89b8dd50569ef1aba3c'), 'zipcode': '91331', 'total_population': 103689, 'male_population': 52358, 'female_population': 51331, 'total_household': 22465, 'city_name': 'Pacoima', 'adjusted_gross_income': 1193295}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9ae'), 'zipcode': '90201', 'total_population': 101279, 'male_population': 50658, 'female_population': 50621, 'total_household': 24104, 'city_name': 'Bell Gardens', 'adjusted_gross_income': 951994}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9cc'), 'zipcode': '90280',

In [43]:
# Script to get Top 10 Zipcodes with highest avg AGI per household[Money amounts are in thousands of dollars]
avg_agi = [
        {'$lookup':{
            'from': "city_zipcode",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "city_data"}
        },
        {'$unwind':"$city_data" },
        {'$lookup':{
            'from': "income_data",
            'localField': "zipcode",
            'foreignField': "zipcode",
            'as': "income_data"}
        },
        {'$unwind':"$income_data" },
        {   
        '$project':{
            'zipcode' : 1,
            'city_name' : "$city_data.city",
            'total_household' : 1,
            'adjusted_gross_income':"$income_data.adjusted_gross_income",
            'avg_agi_per_household':{'$divide': [ '$income_data.adjusted_gross_income',
                                                '$total_household' ] }
        }
    },
        { '$sort': {"avg_agi_per_household": -1 }},
        { '$limit': 10 }
]

In [44]:
# Use AVG AGI script for OC [Money amounts are in thousands of dollars]
for data in (db.oc_data.aggregate(avg_agi)):
    print(data)

{'_id': ObjectId('5dd0a8a78dd50569ef1ac09a'), 'zipcode': '92657', 'total_household': 3777, 'city_name': 'Newport Coast', 'adjusted_gross_income': 1980492, 'avg_agi_per_household': 524.3558379666401}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac08c'), 'zipcode': '92625', 'total_household': 6082, 'city_name': 'Corona Del Mar', 'adjusted_gross_income': 1605589, 'avg_agi_per_household': 263.9902992436698}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac096'), 'zipcode': '92651', 'total_household': 11254, 'city_name': 'Laguna Beach', 'adjusted_gross_income': 2964214, 'avg_agi_per_household': 263.39203838635154}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac0c6'), 'zipcode': '92861', 'total_household': 1967, 'city_name': 'Villa Park', 'adjusted_gross_income': 495737, 'avg_agi_per_household': 252.02694458566344}
{'_id': ObjectId('5dd0a8a78dd50569ef1ac082'), 'zipcode': '92603', 'total_household': 7574, 'city_name': 'Irvine', 'adjusted_gross_income': 1875994, 'avg_agi_per_household': 247.68867177185106}
{'_id': Obje

In [45]:
# Use AVG AGI script for LA [Money amounts are in thousands of dollars]
for data in (db.la_data.aggregate(avg_agi)):
    print(data)

{'_id': ObjectId('5dd0a89b8dd50569ef1ab9a3'), 'zipcode': '90067', 'total_household': 1510, 'city_name': 'Los Angeles', 'adjusted_gross_income': 2446768, 'avg_agi_per_household': 1620.3761589403973}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9af'), 'zipcode': '90210', 'total_household': 8669, 'city_name': 'Beverly Hills', 'adjusted_gross_income': 5294682, 'avg_agi_per_household': 610.7604106586688}
{'_id': ObjectId('5dd0a89b8dd50569ef1aba59'), 'zipcode': '91436', 'total_household': 5552, 'city_name': 'Encino', 'adjusted_gross_income': 3284814, 'avg_agi_per_household': 591.6451729106628}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9b1'), 'zipcode': '90212', 'total_household': 5567, 'city_name': 'Beverly Hills', 'adjusted_gross_income': 2786867, 'avg_agi_per_household': 500.60481408298904}
{'_id': ObjectId('5dd0a89b8dd50569ef1ab9c7'), 'zipcode': '90272', 'total_household': 9212, 'city_name': 'Pacific Palisades', 'adjusted_gross_income': 3680882, 'avg_agi_per_household': 399.57468519322623}
{'_id':

# Report:

We pulled 2010 Population data from Orange County and Los Angeles County to compare their Adjusted Gross Income to see who has the highest amount.

## Data sources:

Orange County: json data
Los Angeles County: csv file
Zipcode City list: beautifulsoup from a website
Income: csv file

### Conclusion:

After loading all data to a Mongodb and did some queries, we found out that Los Angeles has the higher AGI per household, having the highest at $1.6 Million in Zipcode 90067.
Fun fact: checked the zip code at Zillow.com and houses in that area, ranges to $35 Million.